src/components/TextList.js

```diff
@@ -12,6 +12,20 @@ export class TextList extends React.PureComponent {
 
   keyExtractor = (item, index) => `index${index}-id${item.id}`;
 
+  onEndReached = async () => {
+    const { query, fetchMoreData } = this.props;
+
+    if (fetchMoreData) {
+      // if there is a pagination, the end of the list is reached, when no more data is returned
+      // from partially fetching, so we need to check the the data to determine the lists end
+      const { data: moreData } = await fetchMoreData();
+
+      this.setState({ listEndReached: !moreData[query].length });
+    } else {
```

Comment:
this determine the end of lists ?

Reply:
`onEndReached` of a FlatList in React Native is triggered, when a user scrolled at the end: https://reactnative.dev/docs/flatlist#onendreached

We pass this method to the prop to call its logic.
With pagination, the end of a list is reached multiple times, because data gets loaded in partially fetched batches. We have a state variable to save the real end.
If there is a pagination, the real end of the list is reached, when no more data is returned from partially fetching, so we need to check the the data to determine the lists end.
Otherwise, without pagination, the end is reached when the lists end is reached. We do not need further logics and can set the variable.